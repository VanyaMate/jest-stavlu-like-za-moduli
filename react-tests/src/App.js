import React, { useEffect, useState } from 'react';

const App = () => {
    const [ data, setData ]     = useState(null);
    const [ toggle, setToggle ] = useState(false);
    const [ value, setValue ]   = useState('');

    const onClick = function () {
        setToggle((prev) => !prev);
    }

    const onChange = function (event) {
        setValue(event.target.value);
    }

    useEffect(() => {
        setTimeout(() => {
            setData({})
        }, 100)
    }, [])

    return (<div>
        <div data-testid={ 'input-value' }>{ value }</div>
        { toggle && <div data-testid={ "toggle-element" }>Toggled</div> }
        { data && <div style={ { color: 'red' } }>data</div> }
        <h1>Hello</h1>
        <button onClick={ onClick }
                data-testid={ 'toggle-button' }>Click
        </button>
        <input placeholder={ 'input' }
               data-testid={ 'input' }
               value={ value }
               onChange={ onChange }/>
    </div>);
};

export default App;