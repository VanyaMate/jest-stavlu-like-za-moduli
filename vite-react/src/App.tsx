import Users from '@/containers/users/users.tsx';
import { Input } from 'antd';
import React, { useCallback, useState } from 'react';
import Button from './components/ui/buttons/button/button.tsx';


const App: React.FC<React.HTMLAttributes<HTMLDivElement>> = () => {
    const [ state, setState ] = useState<boolean>(false);
    const [ value, setValue ] = useState<string>('');

    const toggle = useCallback(() => {
        setState((prev) => !prev);
    }, [ setState ]);

    const onInput = useCallback<(e: React.ChangeEvent<HTMLInputElement>) => void>((e: React.ChangeEvent<HTMLInputElement>) => {
        setValue(e.target.value);
    }, [ setValue ]);


    return (
        <div>
            <h1>Status: { state.toString() }</h1>
            <Button onClick={ toggle }/>
            <Input value={ value } onChange={ onInput }/>
            <Users/>
        </div>
    );
};

export default App;