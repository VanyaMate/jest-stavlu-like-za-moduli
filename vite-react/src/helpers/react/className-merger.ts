type classNameItem = string | null | undefined;

export const cn = function (...list: classNameItem[]): string {
    return list.filter((className: classNameItem) => !!className).join(' ');
};