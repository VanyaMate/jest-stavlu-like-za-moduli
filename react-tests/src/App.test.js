import { render, screen, fireEvent } from '@testing-library/react';
import { act } from "react-dom/test-utils";
import userEvent from "@testing-library/user-event";
import App from './App';

describe('App', () => {
    test('App test', () => {
        render(<App/>);
        const header = screen.getByText(/hello/i);
        const button = screen.getByRole('button');
        const input  = screen.getByPlaceholderText('input');
        expect(header).toBeInTheDocument();
        expect(button).toBeInTheDocument();
        expect(input).toBeInTheDocument();
        expect(input).toMatchSnapshot();
        screen.debug();
    });

    test('Hello2 is null', async () => {
        render(<App/>);
        const header = await screen.findByText(/data/i, {}, { timeout: 1000 });
        expect(header).toBeInTheDocument();
        expect(header).toHaveStyle({ color: 'red' });
    });

    test('Buttons', () => {
        render(<App/>);
        const button = screen.getByTestId('toggle-button');
        expect(screen.queryByTestId('toggle-element')).toBeNull();
        fireEvent.click(button);
        expect(screen.queryByTestId('toggle-element')).toBeInTheDocument();
        fireEvent.click(button);
        expect(screen.queryByTestId('toggle-element')).toBeNull();
    });

    test('Input', async () => {
        render(<App/>);
        act(() => {
            const input = screen.getByTestId('input');
            expect(input).toBeInTheDocument();
            expect(input).toHaveTextContent('');
            const randomText = Math.random().toString();
            fireEvent.input(input, { target: { value: randomText } })

            const inputValuePlaceholder = screen.getByTestId('input-value');
            expect(inputValuePlaceholder).toHaveTextContent(randomText);
            fireEvent.input(input, { target: { value: 'some text2' } })
            expect(inputValuePlaceholder).toHaveTextContent('some text2');
            userEvent.type(input, 'text');
            expect(inputValuePlaceholder).toHaveTextContent('text');
        })
    });
})

