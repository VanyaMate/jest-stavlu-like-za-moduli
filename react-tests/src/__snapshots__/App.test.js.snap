// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App test 1`] = `
<input
  data-testid="input"
  placeholder="input"
  value=""
/>
`;
